### DISCUSSION TITLE
Use ssh proxy (jump server) to fetch data with Napalm driver

### DISCUSSION THREAD

**Discussion by simo735** [netbox-community/netbox#6547] [Q&A] [answered]
Hello,

I have routers that are reachable only through proxies, I would like to know how to configure those devices to uses proxies rather than a direct SSH connection.

Thank you in advance.

**Comment by candlerb** [comment, marked as answer]
Well, firstly that would depend on what kind of proxy you are talking about.  Is it SOCKS5? ssh to jump host and run ssh command there? Something else?

Let's assume it's SOCKS5.  Then that would depend on Napalm having support for connecting via a SOCKS5 proxy.  I don't see anything in their [documentation](https://napalm.readthedocs.io/en/latest/search.html?q=proxy&check_keywords=yes&area=default) that says it does.  If Napalm can't do it, then Netbox can't either.

You could try spinning up a set of [TCP proxies](https://packages.ubuntu.com/search?keywords=tcp+proxy&searchon=all&suite=focal&section=all), one per device, each listening on different IP addresses or ports, which forward to the target devices.  That's completely external to Netbox of course.

**Reply by simo735** [reply]
@candlerb tnxs for your reply
I don't have a preference between SOCKS5, SSH tunnel or other...The important is to find a way to communicate with the devices which are accessible only via jump servers.
I will try TCP proxies, if it works i'll let you inform.

**Reply by palvan** [reply]
@simo735 We have same issue here and I was wondering if you were able to find a solution?

**Reply by simo735** [reply]
@palvan Unfortunately not yet. We still cannot collect the information of the equipment that can be reached with proxy.
@rskjels the article is interesting, i'll try this solution.

**Comment by rskjels** [comment]
This article was posted on the networktocode slack channel: https://www.oasys.net/posts/netbox-napalm-automation-with-bastion-host/
